Khan/perseus
# BUGFIX: [Number Line] - Tick labels not showing in some exercises

## Summary:
A regression was introduced during the last bugfix ([PR 1695](https://github.com/Khan/perseus/pull/1695)) that caused the tick labels to not show for endpoints when the "Show label ticks" options is deselected. This PR fixes that regression by refactoring how the ticks and their labels are generated.

Issue: LEMS-2532

## Test plan:

1. Launch Storybook
1. Review the [Editor Demo](http://localhost:6006/?path=/story/perseuseditor-editorpage--demo) page
1. Add a Number line widget
   * A number line should show with the default 0 - 10 scale
   * Tick marks and labels are provided in steps of 2 (i.e. 0, 2, 4...)
1. Toggle the "Show label ticks" checkbox off
   * All the labels should now be removed except for the endpoints of zero (0) and ten (10)
1. Enter label values that are different than the endpoints
   * For instance, adding one (1) and seven (7)
   * The endpoint labels are now no longer showing
   * Instead, the newly specified labelled points are showing, along with their associated ticks

## Affected behavior:
### No labels specified
#### Before
![Endpoints - Before](https://github.com/user-attachments/assets/1e51c01a-b764-48dc-8f8d-143dcd8adddd)

#### After
![Endpoints - After](https://github.com/user-attachments/assets/47cbe1b7-834f-4a25-aa85-7375aaed75ae)

### Labels specified
#### Before
![Labels - Before](https://github.com/user-attachments/assets/05461418-4e0a-4fe9-b196-98582358b181)

#### After
![Labels - After](https://github.com/user-attachments/assets/59a676c0-ea05-4716-aa82-edbc25bb59bb)
